cmd/soroban-cli/src/commands/contract/init.rs

```diff
@@ -87,380 +89,412 @@ pub enum Error {
 
 impl Cmd {
     #[allow(clippy::unused_self)]
-    pub fn run(&self) -> Result<(), Error> {
-        println!("ℹ️  Initializing project at {}", self.project_path);
-        let project_path = Path::new(&self.project_path);
-
-        init(
-            project_path,
-            &self.frontend_template,
-            &self.with_example,
-            self.overwrite,
-        )?;
+    pub fn run(&self, global_args: &global::Args) -> Result<(), Error> {
+        let runner = Runner {
+            args: self.clone(),
+            print: print::Print::new(global_args.quiet),
+        };
 
-        Ok(())
+        runner.run()
     }
 }
 
 #[derive(RustEmbed)]
 #[folder = "src/utils/contract-init-template"]
 struct TemplateFiles;
+struct Runner {
+    args: Cmd,
+    print: print::Print,
+}
 
-fn init(
-    project_path: &Path,
-    frontend_template: &str,
-    with_examples: &[String],
-    overwrite: bool,
-) -> Result<(), Error> {
-    // create a project dir, and copy the contents of the base template (contract-init-template) into it
-    create_dir_all(project_path).map_err(|e| {
-        eprintln!("Error creating new project directory: {project_path:?}");
-        e
-    })?;
-    copy_template_files(project_path, overwrite)?;
-
-    if !check_internet_connection() {
-        println!("⚠️  It doesn't look like you're connected to the internet. We're still able to initialize a new project, but additional examples and the frontend template will not be included.");
-        return Ok(());
-    }
-
-    if !frontend_template.is_empty() {
-        // create a temp dir for the template repo
-        let fe_template_dir = tempfile::tempdir().map_err(|e| {
-            eprintln!("Error creating temp dir for frontend template");
-            e
-        })?;
-
-        // clone the template repo into the temp dir
-        clone_repo(frontend_template, fe_template_dir.path())?;
-
-        // copy the frontend template files into the project
-        copy_frontend_files(fe_template_dir.path(), project_path, overwrite)?;
-    }
+impl Runner {
+    fn run(&self) -> Result<(), Error> {
+        let project_path = Path::new(&self.args.project_path);
```

Comment:
just made the change! 

curious how much of an impact this would have in this particular instance since we are creating the directories recursively for the path the user passes in.